Bump rocko

frznlogic opened the pull request:
This PR bumps meta-ivi to rocko release.

Commit:
wayland-ivi-extension: Add support for version 2.1

* Add recipe for wayland-ivi-extensions 2.1 release.
  wayland-ivi-extension 2.1 is preferred in pulsar release.
* Add wayland-protocols to FILES

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

Commit:
Remove gstreamer 1.0 1.2.3:

According to discussions, these recipes are no longer needed.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

Commit:
audiomanagerplugins: Depends on libxml2

Add dependencies for libxml2 for the audiomanagerplugins instead of the
runtime dependencies.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

Commit:
Bump vsomeip: Use version 2.10.10

This fixes a compilation bug using newer gcc versions as well as several
threadlock issues.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

Commit:
pco: Remove uninstalled files

The files in sysconfdir are installed from another package.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

Commit:
distro: gstreamer preferred version updated

Rocko release is shipped with gstreamer 1.12.2, 1.8.3 is no longer available.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

slawr commented:
Hi @frznlogic 

Good to see the rocko work is progressing.

The PR description doesn't say whether this is just showing visibility of a work in progress, or you think these commits are ready for review/merge. I know you said in the BIT call that your own fork was an early snapshot and needed some commit history rewriting. Maybe that applies here as well in the current PR state?

My initial comment is that commit messages are very basic, with most having a "headline" only and no body description. For example the vsomeip patch. There is nothing to give context to why it is needed or what it fixes. Such as a reference to a bug tracker ticket/issue or mailing list discussion. The patch itself has no signed off or reviewed by entries. That occurs sometimes of course when something comes from upstream, but then supporting information is even more required. This might be added as a temporary patch until it is taken upstream for example. Coming to this commit months or years down the line it could be hard to piece together why it exists.

frznlogic commented:
Hi,

I think the history is better in this branch than on the gdp branch to be honest, but yes, you are right, it's lacking details for some of the stuff. I'll be adding it. 

Long story: As for the vsomeip, it's a weird case, the patch was written by me as it fails to compile with the new compilers and I tried finding the maintainers but was unable to find them at that point, so I added the patch temporarily in meta-ivi to make it compilable again and then forgot about it.... 

Short question: Do you have a name/mail I can contact regarding vsomeip?

Commit:
pcl: Add patch for proper initialization:

Some lengths for the memset calls are not properly initialized, this patch is temporarily
added to this repo for the rocko release.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

slawr commented:
>Long story: ..

Thanks for the background. As its your patch I would say add a signed off to the patch and expand the commit msg :)

>Short question: Do you have a name/mail I can contact regarding vsomeip?

With vsomeip I am not sure. I recall there being the two versions from Pelagicore and BMW so it may be a mixed team.  Reference list for maintainers is supposed to be the [project table](https://at.projects.genivi.org/wiki/display/PROJ/Projects+Home) in the Project wiki. That currently says:
>Juergen Gehring – Mail to genivi-maint-vsomeip@genivi.org

Try that. I believe that the OSS team were going through the list of maintainers. You could confirm with @jeremiah and @gunnarx.

frznlogic commented:
Thank you, I managed to find Juergen Gehring. I've sent a PR for vsomeip and added that patch here instead for the time being with commit msg etc :). 

I think pelagicore was at some point part of developing vsomeip but I'm not sure, there's at least no fork in pelagicore.. Thank you for the information however, I'll keep the project table in mind in the future :).

gunnarx commented on meta-ivi/recipes-extended/persistence-common-object/persistence-common-object_1.0.3.bb:
```diff
@@ -20,4 +20,8 @@ inherit autotools-brokensep pkgconfig
 
-FILES_${PN} += "${datadir}"
+do_install_append() {
+    rm -rf ${D}${sysconfdir}
```
Is this safe or could it interact with other packages?  
And why is it needed to do this instead of not touching it?

frznlogic replied:
It should not interact with other packages as it's done on the installation destination dir for that specific package, ie, build/tmp/work/<architecture>/<package>/<version>/$sysconfdir

bitbake fails with an error if there are installed files that are not added to any FILES_x directives.

frznlogic replied:
If you have a better solution I'm all ears :)

frznlogic replied:
I guess the optimal solution would be if only one of the packages actually installed that file though :)

gunnarx commented on meta-ivi-test/recipes-extended/persistence-client-library/persistence-client-library/0001-memset-improperly-initialized-length-of-elements.patch:
```diff
@@ -0,0 +1,309 @@
+diff --git a/test/persistence_client_library_test_file.c b/test/persistence_client_library_test_file.c
```
Me no like... ;-) Why don't we upstream the patch instead?  
(By the way, I have commit access to these repos, just in case the maintainer is not responsive, if that's the issue - and this change seems clear enough.)

gunnarx commented on meta-ivi/conf/distro/poky-ivi-systemd.conf:
```diff
@@ -20,13 +20,13 @@ DISTRO_FEATURES_remove = "x11"
 DISTRO_FEATURES_BACKFILL_CONSIDERED = "sysvinit"
 
 # do not use gstreamer 1.2.3 by default
-PREFERRED_VERSION_gstreamer1.0              ?= "1.10.4"
```
The commit comment talks about 1.8.3 but this is 1.10.4?  (no big deal though - we could let that slip through)

gunnarx commented on meta-ivi/recipes-graphics/wayland/wayland-ivi-extension_2.0.bb:
```diff
@@ -0,0 +1,25 @@
+SUMMARY = "Wayland IVI Extension"
```
Should you git rm also the old 1.13 recipe or do we keep it intentionally?

gunnarx commented:
Please push the vsomeip maintainer to include patch upstream if possible.  I didn't request it in review since I would (reluctantly) accept that patch, more than the  persistence change which I definitely think we should put in the component instead.  But it's a good test if vsomeip can accept quick changes.  If that is updating to new compilers, it should be a safe and good change to make.

commit comments, I think most are quite OK.  But pco: "another package" - which package?

Commit:
PCL: Remove patch applied upstream

The 0001-memset-improperly-initialized-length-of-elements.patch was applied upstream
in the persistence-client-library git repository. Remove the temporary patch and
update the main persistence-client-library bb file to point to that GIT hash.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

Commit:
audiomanagerplugins: Fixed perl regexps

As of perl 5.22.0 regex unescaped { have been moved from deprecated to
generating errors. This caused the configure step to fail as we exceped the $
sign, meaning we wind up with a literal { and not ${foo} being replaced with
whatever content is in ${foo} variable.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

Commit:
weston: Patch to add required API for wayland-ivi-extension

The wayland-ivi-extension 2.1 requires the screen_remove_layer API in weston
which this patch adds.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

Commit:
Name change orion to pulsar:

The current release is named pulsar.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

frznlogic commented:
Patch for vsomeip seems to have been upstreamed. I've fixed the recipe to just bump versions instead.

Commit:
dbus: Fix uninitialized variable

Build fails due to a possibly uninitialized variable. The patch fixes the issue.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

Commit:
node-startup-controller: Add package nsm-dummy

The nsm-dummy package is not created. Let's create it.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

Commit:
Audiomanager tests: Bumped to 7.6

AM 7.6 was bumped and tests need to be bumped forward also.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

Commit:
bblayers.conf.sample: meta-yocto renamed meta-poky

The meta-poky layer was renamed to meta-yocto in rocko release.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

slawr commented:
Ah good you picked up on the meta-yocto renaming to meta-poky. I was going to raise it as a review comment as it had me wondering if you had tested your builds with [TEMPLATECONF](https://github.com/frznlogic/meta-ivi/blob/3e1319763c3273a5b8cedf98b2acc3eb79584928/README.build#L8) set.

gunnarx commented:
> Ah good you picked up on the meta-yocto renaming to meta-poky. I was going to raise it as a review comment as it had me wondering if you had tested your builds with TEMPLATECONF set.

That was the error we saw in CI so I had to check what was up.  Bitbake isn't great here - it just prints "ERROR: Unable to start bitbake server" and you have to dig into logs. :-(

slawr commented:
> That was the error we saw in CI so I had to check what was up. Bitbake isn't great here - it just prints "ERROR: Unable to start bitbake server" and you have to dig into logs. :-(

Weird I thought it coughed at the bitbake parse saying  the layer was unavailable.

Update: I just managed to actually try to kick off a build with the old name in bblayers.conf. I got the following good guidance directly in the terminal so I guess it is a foible of how the CI takes output:
$ time bitbake pulsar-image
ERROR: Unable to start bitbake server
ERROR: Server log for this session (/data/steve/genivi-14/m3sk/y-baseline/build/bitbake-cookerdaemon.log):
--- Starting bitbake server pid 11651 at 2018-02-13 17:34:25.162666 ---
ERROR: Layer directory '/data/steve/genivi-14/m3sk/y-baseline/poky/meta-yocto' does not exist! Please check BBLAYERS in /data/steve/genivi-14/m3sk/y-baseline/build/conf/bblayers.conf

frznlogic commented:
I'll pull this PR in, it builds and runs on all platform but the vexpressa9 which seems to be due to qemu/poky issues rather than us at the moment.

Commit:
wayland-ivi-extension tests: Disable sanitizers

Our cross compilers lack the sanitizers options for our targets. This
patch disables the compiler options which causes compilation errors. In
the long run we should probably try and get the cross compiler compiled
with sanitizers enabled.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

Commit:
runqemu: Use bash instead of sh

The script uses syntaxes that are not POSIX sh compatible, but rather bash.

For this reason, the script is now using bash instead of sh.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>
